providers/src/airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -692,6 +737,12 @@ def read_pod_logs(
             post_termination_timeout=post_termination_timeout,
         )
 
+    @tenacity.retry(stop=tenacity.stop_after_attempt(3), wait=tenacity.wait_exponential(), reraise=True)
+    def get_init_container_names(self, pod: V1Pod) -> list[str]:
+        """Return container names from the POD except for the airflow-xcom-sidecar container."""
+        pod_info = self.read_pod(pod)
```

Comment:
do we _need_ to read the pod again here? or could we just look ta the pod we already have?

Reply:
This method mirror get_container_names. May be it is a task for refactoring in separate PR

Reply:
I'm asking about the new code you are adding though, not the old code we are not changing.

Reply:
And to be clear, again, i'm just asking if we really need the pod read, or no.  I don't know the answer.  But since you are adding the code, you should know (or find out) and we should not just add because it was there in a method we're copy pasting.

Reply:
If we have similar code that performs similar tasks, it might be harder to understand why we’re implementing it differently. I’m not the code owner of podmanager, and I’m open to aligning with your perspective. I just wanted to mention some potential issues.

Reply:
Listen, I welcome your contribution.  That's how new life and new committers come into the airflow community.  But I don't think it's asking too much to ask you to think through whether the code you propose to add is actually needed.  If you can't be bothered to think it through, maybe this is not the project for you.

Indeed it looks like it is not necessary.  Look at line 623 of pod.py.  We read the pod there.  So not sure why you need to read the pod again to get the init container names, unless you think the init containers can change between these two places.

You can't assume that just because the code is in airflow, that means it's good, or that it was thought through.  When we touch code, that's an opportunity to examine it critically.  And when we add new code, that especially is a time to do so.  Otherwise complexity just increases.

Reply:
I don’t have a strong opinion on this. I always focus on making the minimal number of changes to other parts of the code execution flow.

At the same time, I try to write code in a consistent style, following the same patterns as those already in place.

Since I’m not the maintainer of this project, I’m open to any changes that are needed. I trust that you have a better understanding of what needs to be done in this project.

Reply:
It is not required to do kubeapi call.

Fixed